src/torcontrol.cpp

```diff
@@ -752,7 +757,9 @@ void StartTorControl()
         return;
     }
 
-    torControlThread = std::thread(std::bind(&TraceThread<void (*)()>, "torcontrol", &TorControlThread));
+    torControlThread = std::thread(&TraceThread<std::function<void()>>, "torcontrol", [onion_service_target] {
+        TorControlThread(onion_service_target);
```

Comment:
`TorControlThread()` only takes a reference to this object and uses it in another thread. I think this is dangerous because the liftetime of the object may be shorter than the lifetime of the thread. Better make `TorControlThread()`'s argument just `CService` so that the object is copied.

Reply:
wrt properties of `std::thread()` constructor and `TraceThread()` function I don't think it is dangerous. But objects of `CService` are small, so going implement passing by-copy.

Reply:
[Updated](https://github.com/bitcoin/bitcoin/pull/19991#issuecomment-699005931).